### DISCUSSION TITLE
xmake能否设置不配置优化等级

### DISCUSSION THREAD

**Discussion by jiangyunfei1** [xmake-io/xmake#5016] [Q&A]
![image](https://github.com/xmake-io/xmake/assets/40583348/a23c0b0a-6de8-4176-80d7-e2af0a78cd48)
目前我在linux嵌入式平台中从Cmake切换到xmake，出现一个问题。Cmake默认不会设置优化等级，但是xmake默认会设置。但是这个交叉编译工具链在O0下代码运行正常，在其他优化等级O1 Os O2等都不正常。所以release默认下编译运行结果和Cmake不一样，能否像CMake一样不去修改编译工具的优化等级

**Comment by waruqi** [comment]
xmake 默认也不会设置，只不过是默认创建的模版工程，加了 `add_rules("mode.release")` 配置，默认开启使用了 xmake 提供的 release mode 规则。。

你可以完全去掉这个配置，不用 xmake 提供的 release 配置，就不会加了，也可以自己定义 release 规则，或者通过 `set_optimize("none")` 覆盖禁用部分配置。。

如果你的工程什么 add_rules mode 都没加，就一个 target 。。xmake 是不会自动加任何 flags 的
